### PR TITLE
ensuring PPID to be 0 when HOSTPPID is assigned to PPID

### LINKED ISSUE #1388: Identifying kubectl exec
```json
{
  "Timestamp": 1691604077,
  "UpdatedTime": "2023-08-09T18:01:17.064751Z",                                                                                                                                                                                 
  "ClusterName": "default",                                                                                                                                                                                                     
  "HostName": "rh87-demo-node",                                                                                                                                                                                                 
  "NamespaceName": "default",                                                                                                                                                                                                   
  "Owner": {                                                                                                                                                                                                                      
     "Ref": "Pod",                                                                                                                                                                                                                 
     "Name": "ubuntu",                                                                                                                                                                                                             
     "Namespace": "default"                                                                                                                                                                                                      
  },                                                                                                                                                                                                                            
  "PodName": "ubuntu",                                                                                                                                                                                                          
  "Labels": "app=ubuntu",                                                                                                                                                                                                       
  "ContainerID": "3df9822bb4b0f860d9461489a51e8663def11026fac1bd55db705865ab434771",                                                                                                                                            
  "ContainerName": "ubuntu",                                                                                                                                                                                                    
  "ContainerImage": "docker.io/library/ubuntu:latest@sha256:0bced47fffa3361afa981854fcabcd4577cd43cebbb808cea2b1f33a3dd7f508",                                                                                                  
  "ParentProcessName": "/var/lib/rancher/k3s/data/fac8750497b6e8dd01288d91f78fa4730acebd06ffcbfa232e5120b1fec6e476/bin/runc",                                                                                                   
  "ProcessName": "/usr/bin/bash",                                                                                                                                                                                               
  "HostPPID": 3266785,                                                                                                                                                                                                          
  "HostPID": 3266789,                                                                                                                                                                                                           
  "PPID": 3266785,                                                                                                                                                                                                              
  "PID": 36,                                                                                                                                                                                                                    
  "Type": "ContainerLog",                                                                                                                                                                                                       
  "Source": 
  "/var/lib/rancher/k3s/data/fac8750497b6e8dd01288d91f78fa4730acebd06ffcbfa232e5120b1fec6e476/bin/runc",                                                                                                              
  "Operation": "Process",                                                                                                                                                                                                       
  "Resource": "/usr/bin/bash",                                                                                                                                                                                                  
  "Data": "syscall=SYS_EXECVE",                                                                                                                                                                                                 
  "Result": "Passed"
}                                                                                                                                                                                                                             
```

### PR DESCRIPTION
**Pull request by Prateeknandle**
**Purpose of PR?**:
`PPID` inside the container takes `PPID` of runc during `kubectl exec` therefore we add a check when `PPID == HOSTPPID` we make `PPID = 0`

Fixes #1388 

**Does this PR introduce a breaking change?**
No

**If the changes in this PR are manually verified, list down the scenarios covered:**:
tested locally, during kubectl exec `PPID` is 0 and for other process like `ls`, `PPID` has its original value
log for kubectl exec:
```
== Log / 2023-09-13 08:49:05.195945 ==
ClusterName: default
HostName: prateek-lenovo-ideapad-310-15isk
NamespaceName: default
PodName: nginx-86c557dcf4-fs26c
ContainerName: 09c3cbf01faacf6e6e3246200a8341138b2e6aea9ed7876d05f0144092bdc8c1
ContainerID: 09c3cbf01faacf6e6e3246200a8341138b2e6aea9ed7876d05f0144092bdc8c1
Type: ContainerLog
Source: /usr/bin/bash
Operation: Syscall
Data: syscall=SYS_SETUID userid=0
Result: Passed
HostPID: 274313
HostPPID: 274308
PID: 452
PPID: 0
ProcessName: /usr/bin/bash
UID: 0
```

**Additional information for reviewer?** :
1. during `json.Marshal`, due to `omitempty` tag `PPID` with value 0 was not shown in the logs therefore added `sed` command to remove the omitempty tag from the `.pb.go` file.
2. After this PR is merged we also need to update protobuf in kubearmor-client

**Checklist:**
- [ ] Bug fix. Fixes #<issue number>
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update
- [ ] PR Title follows the convention of  `<type>(<scope>): <subject>`
- [ ] Commit has unit tests
- [ ] Commit has integration tests

### CODE REVIEW

File: protobuf/Makefile [hunk 1]
```diff
@@ -15,6 +15,7 @@ go.sum: go.mod
 %.pb.go: %.proto
 	go mod tidy
 	protoc --go_out=. --go_opt=paths=source_relative --go-grpc_out=. --go-grpc_opt=paths=source_relative,require_unimplemented_servers=false $<
+	sed -i 's/\bPPID,omitempty\b/PPID,/' kubearmor.pb.go
```
Comment (nyrahul) [inline protobuf/Makefile]: Please add UID as well.

**Review by nyrahul — DISMISSED** [review]

File: protobuf/kubearmor.pb.go [hunk 2]
```diff
@@ -1,7 +1,7 @@
 // Code generated by protoc-gen-go. DO NOT EDIT.
 // versions:
-// 	protoc-gen-go v1.30.0
-// 	protoc        v4.23.4
+// 	protoc-gen-go v1.26.0
+// 	protoc        v3.19.4
```
Comment (daemon1024) [inline protobuf/kubearmor.pb.go]: Can you use latest protoc? that will avoid these changes latest protoc seems to be v4.23.4

**Review by daemon1024 — APPROVED** [review]
LGTM, one small nit

**Review by daemon1024 — APPROVED** [review]
LGTM 👌🏽 

Do we need to update protobufs on client side to support the zero values? Or will it work OOB.

**Comment by Prateeknandle** [comment]
> Do we need to update protobufs on client side to support the zero values? Or will it work OOB.

Yes we have to update protobufs on client side as well

**Review by daemon1024 — APPROVED** [review]

**Comment by daemon1024** [comment]
Can you please make updates to kubearmor client as well?

**Pull request merged**